test/unit-test/cellular_3gpp_urc_handler_utest.c

```diff
@@ -0,0 +1,564 @@
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file cellular_3gpp_urc_handler.c
+ * @brief Unit tests for functions in cellular_common_internal.h.
+ */
+
+#include <string.h>
+#include <stdbool.h>
+
+#include "unity.h"
+
+#include "cellular_config.h"
+#include "cellular_config_defaults.h"
+
+/* Include paths for public enums, structures, and macros. */
+#include "cellular_platform.h"
+#include "cellular_common_internal.h"
+#include "cellular_common_api.h"
+#include "cellular_types.h"
+
+#include "mock_cellular_at_core.h"
+#include "mock_cellular_common.h"
+
+static int parseRegStatusCase = 0;
+static int parseLacTacInRegCase = 0;
+static int parseCellIdInRegCase = 0;
+static int parseRatInfoInRegCase = 0;
+static int parseRejectTypeInRegCase = 0;
+static int parseRejectCauseInRegCase = 0;
+/* ============================   UNITY FIXTURES ============================ */
+
+/* Called before each test method. */
+void setUp()
+{
+    parseRegStatusCase = 0;
+    parseLacTacInRegCase = 0;
+    parseCellIdInRegCase = 0;
+    parseRatInfoInRegCase = 0;
+    parseRejectTypeInRegCase = 0;
+    parseRejectCauseInRegCase = 0;
+}
+
+/* Called after each test method. */
+void tearDown()
+{
+}
+
+/* Called at the beginning of the whole suite. */
+void suiteSetUp()
+{
+}
+
+/* Called at the end of the whole suite. */
+int suiteTearDown( int numFailures )
+{
+    return numFailures;
+}
+
+/* ========================================================================== */
+
+void MockPlatformMutex_Destroy( PlatformMutex_t * pMutex )
+{
+    pMutex->created = false;
+}
+
+bool MockPlatformMutex_Create( PlatformMutex_t * pNewMutex,
+                               bool recursive )
+{
+    pNewMutex->created = true;
+    return true;
+}
+
+void MockPlatformMutex_Unlock( PlatformMutex_t * pMutex )
+{
+}
+
+void MockPlatformMutex_Lock( PlatformMutex_t * pMutex )
+{
+}
+
+CellularATError_t MOCK_Cellular_ATStrtoi_CALLBACK( const char * pStr,
+                                                   int32_t base,
+                                                   int32_t * pResult,
+                                                   int cmock_num_calls )
+{
+    *pResult = atoi( pStr );
+
+    return CELLULAR_AT_SUCCESS;
+}
+
+CellularATError_t MOCK_Cellular_ATGetNextTok_CALLBACK( char ** ppString,

```

Comment:
Same as in previous file - what do you think about having different small functions?

Reply:
Yeah, that function seems too large, I will separate it into small ones.